flutter_local_notifications/lib/src/notification_details.dart

```diff
@@ -19,3 +31,73 @@ class NotificationDetails {
   /// Notification details for macOS.
   final MacOSNotificationDetails macOS;
 }
+
+class NotificationData {
+  NotificationData(
+    this.id,
+    this.title,
+    this.body,
+    this.scheduledDate,
+    this.platformSpecifics, {
+    this.payload,
+    this.androidAllowWhileIdle = false,
+    this.scheduledNotificationRepeatFrequency,
+    this.uiLocalNotificationDateInterpretation =
+        UILocalNotificationDateInterpretation.absoluteTime,
+    this.platform = const LocalPlatform(),
+  });
+
+  final int id;
+  final String title;
+  final String body;
+  final TZDateTime scheduledDate;
+  final NotificationDetails platformSpecifics;
+  final String payload;
+  final bool androidAllowWhileIdle;
+  final ScheduledNotificationRepeatFrequency
+      scheduledNotificationRepeatFrequency;
+  final UILocalNotificationDateInterpretation
+      uiLocalNotificationDateInterpretation;
+  final Platform platform;
+
+  Map<String, dynamic> get _platformSpecifics {
+    Map<String, dynamic> _specs;
+
+    if (platform.isAndroid) {
+      _specs = platformSpecifics?.android?.toMap();
+    }
+    if (platform.isIOS) {
+      _specs = platformSpecifics?.iOS?.toMap();
+    }
+    if (platform.isMacOS) {
+      _specs = platformSpecifics?.macOS?.toMap();
+    }
+    return _specs ?? <String, dynamic>{};
+  }
+
+  Map<String, dynamic> toMap() {
```

Comment:
All mapping code has been moved to be extension methods (e.g. in `method_channel_mappers.dart`. Can you move this code as well? The reason for doing this was to not expose `toMap()` in the public API surface as it's only used internally